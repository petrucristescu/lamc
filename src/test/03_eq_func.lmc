@b_v1 false
@b_v2 true

~foo (
    print "foo called"
)

~bar (
    print "bar called"
)

~choose_fn x,y (
    eq x y foo bar
)

~(
    print v1
    print v2
    print(eq 1 1 foo bar)
    print(eq 1 2 foo bar)

    print choose_fn 1 1
    print choose_fn 1 2

    print choose_fn v1 v1
    print choose_fn v1 v2
)