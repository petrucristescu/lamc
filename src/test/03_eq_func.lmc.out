false
true
foo called
bar called
foo called
bar called
bar called
bar called
